### ISSUE TITLE
Bayesian Optimization and Surprise 

### ISSUE THREAD

**Issue by NaveganteX** [NicolasHug/Surprise#298]
Hello Nicolas! 

I wanted to know if it possible to merge Surprise and the library called bayes_opt in order to apply Bayesian Optimization for parameters tuning. Maybe by doing so, I can reduce computation time taken by GridSearchCV() from your library. 

Thank you.

Regards.

#### Steps/Code to Reproduce

```
tuning_params = dict()
tuning_params = { 
  ’n_factors’: [50,250]
  "lr_all": [0.001, 0.009]
 }

def BO_func(k, learning_rate):
     algo = SVD()
     algo.fit(trainset)
     predictions = algo.test(testset)

  return accuracy.rmse(predictions)

from bayes_opt import BayesianOptimization

optimizer = BayesianOptimization(
  f = BO_func,
  pbounds = tuning_params,
  verbose = 5,
  random_state = 5, 
 )

optimizer.maximize(
  init_points = 4,
  n_iter = 3, 
 )
